vrecord

```diff
@@ -37,6 +37,24 @@ MPVOPTS+=(--load-scripts=no)
 MPVOPTS+=(--script "${SCRIPTDIR}/qcview.lua")
 MPVOPTS+=(--really-quiet)
 
+# Set variables for system differences
+if [[ "$(uname -s)" = "Darwin" ]] ; then
+    OS_TYPE="macOS"
+    DEFAULTFONT='/Library/Fonts/Andale Mono.ttf'
+    HARDWARE_REPORT=$(system_profiler SPHardwareDataType)
+    CORE_COUNT="$(_parse_report "Total Number of Cores" "${HARDWARE_REPORT}")"
+    OPEN_COMMAND="open"
+elif [[ "$(uname -s)" = "Linux" ]] ; then
+    OS_TYPE="linux"
+    DEFAULTFONT='/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf'
+    HARDWARE_REPORT=$(lscpu)
+    CORE_COUNT="$(nproc --all)"
+    OPEN_COMMAND="xdg-open"
+else
+    echo 'Unsupported OS detected. Exiting.'
```

Comment:
If you're going to exit here, you should include the output of `$(uname -s)` so the user gets a clue.

Reply:
makes sense!